plugin gone from sidebar

i used this plugin for a while and it was great but one day it just wasnt in the sidebar anymore, if i search for the plugin its still installed i just cant see anything its tracking without it in sidebar. ive tried to delete it and reinstall but got the same problem. i havent found anyone else with the same problem as me so i have no idea what to do, if you have any idea how i can get it to work again i would appreciate it a lot!

1) Error injecting constructor, com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
  at com.tobmistaketracker.panel.TobMistakeTrackerPanel.<init>(Unknown Source)
  at com.tobmistaketracker.panel.TobMistakeTrackerPanel.class(Unknown Source)
  while locating com.tobmistaketracker.panel.TobMistakeTrackerPanel

1 error
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:424)
	at net.runelite.client.plugins.config.PluginListPanel.startPlugin(PluginListPanel.java:310)
	at net.runelite.client.plugins.config.PluginListItem.lambda$new$4(PluginListItem.java:178)
	at java.desktop/javax.swing.AbstractButton.fireItemStateChanged(Unknown Source)
	at java.desktop/javax.swing.AbstractButton$Handler.itemStateChanged(Unknown Source)
	at java.desktop/javax.swing.DefaultButtonModel.fireItemStateChanged(Unknown Source)
	at java.desktop/javax.swing.JToggleButton$ToggleButtonModel.setSelected(Unknown Source)
	at java.desktop/javax.swing.AbstractButton.setSelected(Unknown Source)
	at net.runelite.client.plugins.config.PluginListItem.setPluginEnabled(PluginListItem.java:211)
	at net.runelite.client.plugins.config.PluginListPanel.lambda$refresh$5(PluginListPanel.java:245)
	at java.base/java.util.ArrayList.forEach(Unknown Source)
	at net.runelite.client.plugins.config.PluginListPanel.refresh(PluginListPanel.java:240)
	at net.runelite.client.plugins.config.PluginListPanel.rebuildPluginList(PluginListPanel.java:229)
	at net.runelite.client.plugins.config.ConfigPlugin.startUp(ConfigPlugin.java:84)
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:402)
	at net.runelite.client.plugins.PluginManager.lambda$startPlugins$2(PluginManager.java:247)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
  at com.tobmistaketracker.panel.TobMistakeTrackerPanel.<init>(Unknown Source)
  at com.tobmistaketracker.panel.TobMistakeTrackerPanel.class(Unknown Source)
  while locating com.tobmistaketracker.panel.TobMistakeTrackerPanel

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1028)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1054)
	at com.tobmistaketracker.TobMistakeTrackerPlugin.startUp(TobMistakeTrackerPlugin.java:125)
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:402)
	... 28 common frames omitted
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
	at com.google.gson.Gson.fromJson(Gson.java:927)
	at com.tobmistaketracker.state.MistakeStateReader.read(MistakeStateReader.java:40)
	at com.tobmistaketracker.panel.TobMistakeTrackerPanel.<init>(TobMistakeTrackerPanel.java:115)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at com.google.inject.internal.DefaultConstructionProxyFactory$ReflectiveProxy.newInstance(DefaultConstructionProxyFactory.java:126)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:111)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:268)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1019)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1015)
	... 31 common frames omitted
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:385)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
	... 50 common frames omitted

This seems to be all i can find in there that says anything about the mistake tracker plugin.  Idk if any of this is useful If you want to see the full thing i could attatch it for you.

```
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
at com.google.gson.Gson.fromJson(Gson.java:927)
at com.tobmistaketracker.state.MistakeStateReader.read(MistakeStateReader.java:40)
```

Looks like the issue is coming from trying to deserialize the tracked mistakes `.json` file. My guess is that it's malformed somehow (either through manual editing or somehow somewhere something broke).

If you don't mind, could you paste or attach your `mistake-state.txt` file? To get there, if you're in the same directory as the logs you were just looking at, going up one directory should put you in the `.runelite` folder, and you should see folder named `tob-mistake-tracker`. If you go in there, you should see the specified file.

If you don't care about your past tracked mistakes, you can just delete that file and then things should work again, but your tracked mistakes will be reset. If you do care about your past tracked mistakes, then we can take a look at the current content and figure out what's malformed and fix it.

-- 

In the future, I should make sure to catch `JsonSyntaxException` too and recover in some way so the plugin still works.

--- a/README.md
+++ b/README.md
@@ -41,6 +41,9 @@ Coming Soon:
 
 ## Changes
 
+#### 2.2
+* Fix issue where malformed mistake-state file would cause plugin to not load
+
 #### 2.1
 * Add config for toggling mistake overhead text
 

--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,7 @@ repositories {
 	mavenCentral()
 }
 
-def runeLiteVersion = '1.7.15'
+def runeLiteVersion = '1.8.2'
 
 dependencies {
 	compileOnly group: 'net.runelite', name:'client', version: runeLiteVersion
@@ -27,7 +27,7 @@ dependencies {
 }
 
 group = 'com.tobmistaketracker'
-version = '2.1'
+version = '2.2'
 sourceCompatibility = '1.8'
 
 tasks.withType(JavaCompile) {

--- a/src/main/java/com/tobmistaketracker/state/MistakeStateReader.java
+++ b/src/main/java/com/tobmistaketracker/state/MistakeStateReader.java
@@ -1,6 +1,7 @@
 package com.tobmistaketracker.state;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
 import com.google.gson.stream.JsonReader;
 import lombok.extern.slf4j.Slf4j;
 import net.runelite.http.api.RuneLiteAPI;
@@ -38,9 +39,12 @@ public class MistakeStateReader {
             try (BufferedReader reader = Files.newBufferedReader(mistakeStateFilePath);
                  JsonReader jsonReader = new JsonReader(reader)) {
                 MistakeStateManager mistakeStateManager = GSON.fromJson(jsonReader, MistakeStateManager.class);
+                if (mistakeStateManager == null) {
+                    return new MistakeStateManager(mistakeStateWriter);
+                }
                 mistakeStateManager.setMistakeStateWriter(mistakeStateWriter);
                 return mistakeStateManager;
-            } catch (IOException e) {
+            } catch (IOException | JsonSyntaxException e) {
                 log.error("Unable to read mistake state from " + mistakeStateFilePath, e);
             }
         }

